tests/test_def_use_graph.py

```diff
@@ -0,0 +1,61 @@
+import networkx
+import nose
+
+from unittest import mock
+
+from angr.analyses.reaching_definitions.dataset import DataSet
+from angr.analyses.reaching_definitions.def_use_graph import *
+
+
+def test_def_use_graph_has_a_default_graph():
+    def_use_graph = DefUseGraph()
+
+    nose.tools.assert_equal(isinstance(def_use_graph.graph, networkx.DiGraph), True)
+
+
+def test_def_use_graph_refuses_to_instanciate_with_an_inadequate_graph():
+    a_graph = networkx.DiGraph([(1, 2)])
+
+    with nose.tools.assert_raises(ValueError) as init_result:
+        def_use_graph = DefUseGraph(a_graph)
+
+    nose.tools.assert_equal("%s" % init_result.exception, 'In a DefUseGraph, nodes need to be <Definition>s.')
```

Comment:
1) we should not be comparing exception text in tests
2) the error should probably be a typeerror since it is a type error

Reply:
1. ...Why?
2. Argh, right.

back to 1:
I want to test that a `TypeError` is raised with a meaningful message.
I felt like checking the message contained in the exception was a way to ensure that.

I would agree with 1 if there were more precise errors, but it felt silly (over-engineering) to create two errors, inheriting from `TypeError`, for the two cases encountered in this class...

Reply:
1. Oh, right, I abused the `%` formatting to get the message out of the error.

After a bit of research, it seems that the `message` attribute has been deprecated, and it is common to rely on `__str__` to get the message...

Reply:
Echo @rhelmot: Please do not compare exception text in tests. Exceptions are only meant to provide human-readable information and the text may change without making tests fail. Please create a new type of exception if you want to verify that a certain exception is raised.

Also, don't call `__str__()` unless you really have to. Just use `str(xxx)`.